### PR TITLE
Add documentation on helm3 + migration per skyscrapers/engineering#311

### PR DESCRIPTION
**Pull request by simonrondelez**
This PR adds documentation on:
- how to use helm3 in Concourse deployments
- how to migrate your personal helm2 configuration
- how to migrate deployments on your K8s cluster

### CODE REVIEW

File: kubernetes/helm.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+In order to migrate from Helm v2 to v3 all deployments need to be migrated to their respective namespaces.
+This can be done with the [helm-2to3](https://github.com/helm/helm-2to3) plugin:
+
+This post from Helm explains how to set up helm v3 and how to migrate your config and the Helm v2 releases:
+
+- [Official documentation on the v2-v3 migration](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/)
+
+### Breaking changes
+
+Helm v2 charts are mostly compatible with version 3. There are however some things you need to be aware from:
```
Comment (iuriaranda) [inline kubernetes/helm.md:19]: [...] be aware of:

File: Concourse/README.md [hunk 2]
```diff
@@ -327,3 +326,44 @@ jobs:
...
+    source:
+      repository: typositoire/concourse-helm3-resource
+resources:
+  - name: example-helm
+    type: helm
+    source:
+      cluster_url: https://example.eks.amazonaws.com/
+      cluster_ca: ((HELM_CI.cluster_ca))
+      token: ((HELM_CI.token))
+      helm_history_max: 10
```
Comment (duboisph) [inline Concourse/README.md:350]: I suggest we add a note marking the importance of setting `helm_history_max`

File: kubernetes/helm.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Helm
+
+## Setting up Helm v3
+
+- [Install the latest helm binary](https://github.com/helm/helm#install)
+  - Tip: if you want to have both helm v2 and helm v3: rename the binary of one or both versions.
+
+## migration from Helm v2 to v3
```
Comment (duboisph) [inline kubernetes/helm.md:8]: migration -> Migration :)

File: kubernetes/helm.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+- [Official documentation on the v2-v3 migration](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/)
+
+### Breaking changes
+
+Helm v2 charts are mostly compatible with version 3. There are however some things you need to be aware of:
+
+- [Tiller is removed](https://v3.helm.sh/docs/faq/#removal-of-tiller)
+Once the migration of all the deployments is done we will remove Tiller from our clusters.
+- [the way CRDs are handled by Helm has changed](https://helm.sh/docs/topics/charts/#custom-resource-definitions-crds)
```
Comment (duboisph) [inline kubernetes/helm.md:23]: the -> The

Comment (duboisph) [reply kubernetes/helm.md:23]: Sidenote: also not really an issue for customers, they don't manage CRDs :)

File: kubernetes/helm.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+The way Helm 3 handles CRDs has been changed. They are now treated as special resources and are never upgraded by Helm once installed. The upgrade operation should be done by the cluster operators (admins) with extra care. Few things about this change:
+>
+> - CRDs should be put in the crds directory at the top level of chart directory.
+> - The YAML files in this directory cannot be templatized like any other resources from templates directory.
+> - The crd-install hook , which used to take care of installing CRD files from templates directory has been removed. If there are any files with crd-install annotation, then those are skipped by Helm.
+> - Files from crds directory are applied first before rendering the chart. These are never applied again if the CRDs exist in the cluster.
+
+- [Releases are now kept in their respective K8s Namespace](https://v3.helm.sh/docs/faq/#release-names-are-now-scoped-to-the-namespace)
+With this change its not needed anymore to add an environment parameter to your deployment. The names only need to be unique per K8s Namespace.
```
Comment (duboisph) [inline kubernetes/helm.md:32]: its -> it is (or it's)

File: kubernetes/helm.md (hunk 4 above)
Comment (simonrondelez) [reply kubernetes/helm.md:23]: true. However i added it for completeness. Should i remove it?

Comment (duboisph) [reply kubernetes/helm.md]: No let's keep it, it's indeed good for completeness.

**Pull request merged**